### PR TITLE
Fix spectating for LTS

### LINKED ISSUE #501: `ns_allow_spectators` doesn't work properly in LTS
When creating a server with `+ns_allow_spectators 1`, and toggling spectate in a private match, the spectators end up getting stuck spectating at the bottom of the map:
image

This might only be the issue for game modes like LTS, that have titan selection screens

### PR DESCRIPTION
**Pull request by L1ghtman2k**
This change by @Khalmee adds spectating to LTS, and fixes https://github.com/R2Northstar/NorthstarMods/issues/501.

This might not be the best place to put the fix in, so feel free to change things around if needed

### CODE REVIEW

File: Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut [hunk 1]
```diff
@@ -177,17 +177,23 @@ void function GameStateEnter_PickLoadout()
...
 
+    //added by Khalmee for the sake of fixing LTS spectator mode
+    foreach(player in GetPlayerArray()){
+        if(IsPrivateMatchSpectator(player))
+        InitialisePrivateMatchSpectatorPlayer( player )
+    }
+    //end of Khalmee changes
+
+    SetGameState( eGameState.Prematch )
+}
```
Comment (GeckoEidechse) [inline Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:196]: Is it just me or did the indentation on this line change? Might be an accidental tabs/spaces edit ^^

Comment (RoyalBlue1) [reply Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:196]: since all of it is spaces now it needs to be changed to tabs anyway

Comment (L1ghtman2k) [reply Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut]: replaced back to tabs

File: Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut [hunk 2]
```diff
@@ -184,11 +184,17 @@ void function GameStateEnter_PickLoadout_Threaded()
 	// titan selection menu can change minPickLoadOutTime so we need to wait manually until we hit the time
 	while ( Time() < GetServerVar( "minPickLoadOutTime" ) )
 		WaitFrame()
-	
+
+	//added by Khalmee for the sake of fixing LTS spectator mode
+	foreach(player in GetPlayerArray()){
+		if(IsPrivateMatchSpectator(player))
+		InitialisePrivateMatchSpectatorPlayer( player )
+	}
```
Comment (GeckoEidechse) [inline Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:192]: Formatting
```suggestion
	foreach ( player in GetPlayerArray() ){
		if ( IsPrivateMatchSpectator( player ) )
		InitialisePrivateMatchSpectatorPlayer( player )
	}
```

Comment (GeckoEidechse) [reply Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:192]: Wait, why can I not commit suggestion myself? :c

Comment (L1ghtman2k) [reply Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut]: just invited you to the fork repo, should be able to commit suggestions now

File: Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut [hunk 3]
```diff
@@ -184,11 +184,17 @@ void function GameStateEnter_PickLoadout_Threaded()
 	// titan selection menu can change minPickLoadOutTime so we need to wait manually until we hit the time
 	while ( Time() < GetServerVar( "minPickLoadOutTime" ) )
 		WaitFrame()
-	
+
+	//added by Khalmee for the sake of fixing LTS spectator mode
+	foreach ( player in GetPlayerArray() ){
+		if ( IsPrivateMatchSpectator( player ) )
+		InitialisePrivateMatchSpectatorPlayer( player )
+	}
```
Comment (ASpoonPlaysGames) [inline Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut:192]:
```suggestion
	foreach ( player in GetPlayerArray() )
	{
		if ( IsPrivateMatchSpectator( player ) )
			InitialisePrivateMatchSpectatorPlayer( player )
	}
```

**Comment by ASpoonPlaysGames** [comment]
I'll take a proper look at the scripts later today (after work)

I find it weird that we aren't already doing `InitialisePrivateMatchSpectatorPlayer`

**Review by ASpoonPlaysGames — CHANGES_REQUESTED** [review]
This should probably be moved over to `GameStateEnter_Prematch` (`scripts/vscripts/mp/_gamestate_mp.nut`)

Example:
```
// eGameState.Prematch
void function GameStateEnter_Prematch()
{	
	int timeLimit = GameMode_GetTimeLimit( GAMETYPE ) * 60
	if ( file.switchSidesBased )
		timeLimit /= 2 // endtime is half of total per side
	
	SetServerVar( "gameEndTime", Time() + timeLimit + ClassicMP_GetIntroLength() )
	SetServerVar( "roundEndTime", Time() + ClassicMP_GetIntroLength() + GameMode_GetRoundTimeLimit( GAMETYPE ) * 60 )
	
	if ( !GetClassicMPMode() && !ClassicMP_ShouldTryIntroAndEpilogueWithoutClassicMP() )
		thread StartGameWithoutClassicMP()

	// Initialise any spectators. Hopefully they are all initialised already in CodeCallback_OnClientConnectionCompleted
	// (_base_gametype_mp.gnut) but for modes like LTS this doesn't seem to happen late enough to work properly.
	foreach ( player in GetPlayerArray() )
	{
		if ( IsPrivateMatchSpectator( player ) )
			InitialisePrivateMatchSpectatorPlayer( player )
	}
}
```

**Comment by L1ghtman2k** [comment]
The changes are relatively small, can we recreate this PR, and use a branch instead? Forking across github orgs turned out to be a great pain

**Comment by ASpoonPlaysGames** [comment]
ohhhhh yeah i didn't realise you used your main branch for this.

yeah feel free to recreate the PR on a different branch

**Comment by L1ghtman2k** [comment]
@GeckoEidechse 
@ASpoonPlaysGames 
You should be able to change it now, I moved it to my personal repo

**Comment by GeckoEidechse** [comment]
Note to self, on merge, add

```
Co-authored-by: Khalmee <87766895+Khalmee@users.noreply.github.com>
```

File: Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut (hunk 3 above)
Comment (GeckoEidechse) [reply Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut]: Addressed in c794c84753d94b0484d6772cab38ee71c0c2d881

**Review by GeckoEidechse — APPROVED** [review]
Confirmed fixes #501 in testing.

Will merge the moment someone gives code review :D

**Comment by ASpoonPlaysGames** [comment]
> This should probably be moved over to `GameStateEnter_Prematch` (`scripts/vscripts/mp/_gamestate_mp.nut`)

I have given code review, just waiting on changes

**Comment by GeckoEidechse** [comment]
> > This should probably be moved over to `GameStateEnter_Prematch` (`scripts/vscripts/mp/_gamestate_mp.nut`)
> 
> I have given code review, just waiting on changes

whoops, I overlooked that, my bad, will do the change myself rq ^^

**Review by GeckoEidechse — APPROVED** [review]
Reconfirmed working in testing.

**Review by ASpoonPlaysGames — APPROVED** [review]
Code looks good :)

**Pull request merged**